src/Raven.Client/Documents/IDocumentStore.cs

```diff
@@ -171,5 +171,7 @@ public interface IDocumentStore : IDisposalNotification
         DatabaseSmuggler Smuggler { get; }
 
         IDisposable SetRequestTimeout(TimeSpan timeout, string database = null);
+
+        void AssertInitialized();
```

Comment:
I don't like it on the root interface. The user will never call it.
Better to do downcast when needed.